cmd/csaf_provider/create.go

```diff
@@ -87,20 +90,85 @@ func createFeedFolders(c *config, wellknown string) error {
 // and writes the CSAF field inside the file.
 func createSecurity(c *config, wellknown string) error {
 	security := filepath.Join(wellknown, "security.txt")
-	if _, err := os.Stat(security); err != nil {
+
+	path := fmt.Sprintf(
+		"%s/.well-known/csaf/provider-metadata.json",
+		c.CanonicalURLPrefix)
+
+	st, err := os.Stat(security)
+	if err != nil {
 		if os.IsNotExist(err) {
 			f, err := os.Create(security)
 			if err != nil {
 				return err
 			}
-			fmt.Fprintf(
-				f, "CSAF: %s/.well-known/csaf/provider-metadata.json\n",
-				c.CanonicalURLPrefix)
+			fmt.Fprintf(f, "CSAF: %s\n", path)
 			return f.Close()
 		}
 		return err
 	}
-	return nil
+
+	// Load it line wise
+	found, lines, err := func() (bool, []string, error) {
+		f, err := os.Open(security)
+		if err != nil {
+			return false, nil, err
+		}
+		defer f.Close()
+		var lines []string
+		sc := bufio.NewScanner(f)
+		for sc.Scan() {
+			line := sc.Text()
+			if s := strings.TrimLeftFunc(line, unicode.IsSpace); strings.HasPrefix(s, "CSAF:") {
+				// Check if we are already in.
+				if strings.TrimSpace(s[len("CSAF:"):]) == path {
+					return true, nil, nil
+				}
+			}
+			lines = append(lines, line)
+		}
+		return false, lines, sc.Err()
+	}()
+	if err != nil {
+		return err
+	}
+
+	// we are already in the file.
+	if found {
+		return nil
+	}
+
+	// Insert our CSAF line at the beginning
+	// to get higher priority over possible existing CSAF lines.
+	csafLine := fmt.Sprintf("CSAF: %s", path)
+	lines = append([]string{csafLine}, lines...)
```

Comment:
The examples in https://datatracker.ietf.org/doc/html/draft-foudil-securitytxt-12 suggest that an additional linefeed could make the file more readable. In addition we could consider adding a comment like:

```
# proposed CSAF-2.0 field. Inserted by csaf_provider/create on 2022-04-22
```

Reply:
Could be done in separate PR.

Reply:
Added the newline for now.